### PR TITLE
[FLOSS T1] fix-spelling: Fix an issue where tests fails, when path contains whitespaces

### LINKED ISSUE #4672: [FLOSS T1] scripts/sed is broken
## Steps to Reproduce the Problem

See https://build.libelektra.org/blue/organizations/jenkins/libelektra/detail/PR-4246/18/tests for example

## Expected Result

`scripts/sed` is able to handle spaces in the path. After all that's why we use `/build directory/` on Jenkins...

## Actual Result

```
sed: can't read ./build: No such file or directory
sed: can't read directory/src/bindings/jna/plugins/sorted/README.md: No such file or directory
sed: can't read ./build: No such file or directory
sed: can't read directory/src/bindings/jna/plugins/linkchecker/README.md: No such file or directory
sed: can't read ./build: No such file or directory
sed: can't read directory/src/bindings/jna/plugins/whitelist/README.md: No such file or directory
sed: can't read ./build: No such file or directory
sed: can't read directory/src/bindings/jna/README.md: No such file or directory
sed: can't read ./build: No such file or directory
sed: can't read directory/src/bindings/jna/libelektra-kotlin/README.md: No such file or directory
sed: can't read ./build: No such file or directory
sed: can't read directory/src/tools/hub-zeromq/README.md: No such file or directory
```

## System Information

- Jenkins CI

## Further Notes

PS. I'd also rename `scripts/sed` to e.g. `scripts/spelling.sed`, because there could be other sed scripts too.

### PR DESCRIPTION
**Pull request by Bujuhu**
- Fix an issue, where scripts/dev/fix-spelling won't run correctly, when the path contains whitespaces
- This also fixes the failing check_spelling tests on jenkins
- Fixes #4672
- Renamed sed to the more verbose spelling.sed
- Add new filters, to ignore markdown files in some external dependencies

## Basics

The problem was, that the output of `find | egrep` wasn't parsed to `sed` correctly, causing sed to identify arguments incorrectly. We've changed it by filtering the files directly using `find`. This also has the benefit that we can get rid of the deprecated `egrep` command, which was printing warnings before.

Additionally, we expanded the filter to exclude other external dependencies that might show up during build. Namely, those from npm packages and within googletest-src

We also renamed the sed file to spelling.sed as suggested in #4672
We looked at the project and didn't find any other references to the file than in /scripts/dev/fix-spelling (other than the open PRs #4692and #4646 which should probably be merged first to make it easier to handle merge conflicts), so we don't believe this would introduce any new issues.

- We reproduced the problem by renaming the build folder to something that includes a whitespace and then try to run the `scripts/dev/fix-spelling`, which resulted in the same issue as described in #4672
- the test is running on jenkins
- we implemented this in pair programming (or rather, pair debugging)

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [x] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [x] The PR is rebased with current master.

## Checklist

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation (see [Documentation Guidelines](https://www.libelektra.org/devgettingstarted/documentation))
- [ ] I fixed all affected decisions (see [Decision Process](https://www.libelektra.org/decisions/decision-process))
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://www.libelektra.org/devgettingstarted/coding))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned [every code](/.reuse/dep5) not directly written by me in [reuse syntax](https://reuse.software/)

## Review

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

- [ ] Add the "work in progress" label if you do not want the PR to be reviewed yet.
- [ ] Add the "ready to merge" label **if the basics are fulfilled** and no further pushes are planned by you.

### CODE REVIEW

**Comment by Bujuhu** [comment]
Also, on another note, I'm not quite sure why the _co-authored-by_ isn't correctly rendered by GitHub, if someone knows what the issue is, please let me know!

**Comment by kodebach** [comment]
> Also, on another note, I'm not quite sure why the _co-authored-by_ isn't correctly rendered by GitHub, if someone knows what the issue is, please let me know!

AFAIK [it has to be a commit trailer](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors). So it must be at the end of the commit message with _two_ empty lines before.

**Comment by Bujuhu** [comment]
> > Also, on another note, I'm not quite sure why the _co-authored-by_ isn't correctly rendered by GitHub, if someone knows what the issue is, please let me know!
> 
> AFAIK [it has to be a commit trailer](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors). So it must be at the end of the commit message with _two_ empty lines before.

Thank you!

**Comment by flo91** [comment]
jenkins build libelektra please

**Review by joni1993 — APPROVED** [review]
**Documentation is introductory, concise, good to read and describes everything what the PR does**
Yes
**Examples are well chosen and understandable**
N/A
**Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)**
Yes, shfmt is applied.
**APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)**
N/A
Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)
N/A

**Comment by markus2330** [comment]
jenkins build libelektra please

File: doc/news/_preparation_next_release.md [hunk 1]
```diff
@@ -415,7 +415,7 @@ This section keeps you up-to-date with the multi-language support provided by El
 
 ## Tests
 
-- <<TODO>>
+- Fix an Issue where scripts/dev/fix-spelling does not work, if a resolved path contains whitespaces _(Juri Schreib @Bujuhu)_ _(Nikola Prvulovic @Dynamichost96)_
```
Comment (markus2330) [inline doc/news/_preparation_next_release.md:418]: Please also write that the file was renamed. Please also fix occurrence within the same file (`https://master.libelektra.org/scripts/sed`). Also doc/news/2022-03-10_0.9.9.md and doc/news/2017-12-22_0.8.21.md and tests/shell/check_spelling.sh have references.

**Review by markus2330 — CHANGES_REQUESTED** [review]

**Comment by Bujuhu** [comment]
jenkins build libelektra please

**Comment by markus2330** [comment]
Great work!

**Pull request merged**